src/core/dynamic_batch_scheduler.cc

```diff
@@ -284,14 +283,7 @@ DynamicBatchScheduler::SchedulerThread(
           }
 
           if (preserve_ordering_) {
-            // There is runner processing payloads before current runner
-            if (last_processing_runner_id_ != -1) {
-              completion_promise =
-                  completion_promises_[last_processing_runner_id_];
-            }
-            last_processing_runner_id_ = runner_id;
-            completion_promises_[runner_id] =
-                std::make_shared<std::promise<void>>();
+            runner_queue_.push(runner_id);
```

Comment:
This should be moved to line 318 since we need to preserve ordering across instances even when not using the dynamic batcher

runner_queue is not protected here by completion_queue_mtx

Reply:
Fixed